### PR TITLE
Adding RAI examples

### PR DESCRIPTION
**Pull request by JoanGi**
This PR enhances the 1.0. BigCode the stack dataset with some RAI properties and adds the DICES-350 dataset to 1.0 as another example of using RAI properties.

### CODE REVIEW

File: datasets/1.0/bigcode-the-stack/metadata.json [hunk 1]
```diff
@@ -47,23 +45,47 @@
...
+  ],
+  "rai:dataSocialImpact": "The Stack is released with the aim to increase access, reproducibility, and transparency of code LLMs in the research community. We expect code LLMs to enable people from diverse backgrounds to write higher quality code and develop low-code applications. Mission-critical software could become easier to maintain as professional developers are guided by code-generating systems on how to write more robust and efficient code. While the social impact is intended to be positive, the increased accessibility of code LLMs comes with certain risks such as over-reliance on the generated code and long-term effects on the software development job market.",
+  "rai:dataSensitive": "The released dataset may contain sensitive information such as emails, IP addresses, and API/ssh keys that have previously been published to public repositories on GitHub. Deduplication has helped to reduce the amount of sensitive data that may exist. The PII pipeline for this dataset is still a work in progress. Researchers who wish to contribute to the anonymization pipeline of the project can apply to join here: https://www.bigcode-project.org/docs/about/join/.",
```
Comment (benjelloun) [inline datasets/1.0/bigcode-the-stack/metadata.json:87]: rai:sensitivePersonalInfo ?

File: datasets/1.0/dices-350/metadata.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    "url": "https://github.com/google-research-datasets/dices-dataset/"
+  },
+  "datePublished": "2024-02-01",
+  "rai:dataCollection": "The input data was sampled from an 8K multi-turn conversation corpus (comprising 48K turns in total) generated by human agents interacting with a generative AI-chatbot. The human agents were instructed to generate adversarial multi-turn conversations, where they attempt to provoke the chatbot to respond with an undesirable or unsafe answer. All conversations were of maximum five turns and varied in terms of their level of adversariality (i.e., degree of harm) and topics.A subset of the conversations (DICES-350) were annotated with gold safety labels (from trust and safety experts) and all conversations with platinum safety labels (from a diverse rater crowd)",
+  "rai:dataCollectionTypeOthers": "DICES-350 consists of 350 adversarial multi-turn conversations,annotated by a pool of annotators along 16 safety criteria.\"",
+  "rai:dataCollectionRaw": "The input data for this data collection was sampled from an 8K multi-turn conversation corpus (comprising 48K turns in total) generated by human agents interacting with a generative AI-chatbot.",
```
Comment (benjelloun) [inline datasets/1.0/dices-350/metadata.json:56]: rai:dataCollectionRawData ?

**Review by benjelloun — COMMENTED** [review]
Can you please check that attributes are named consistently with the published spec?

https://mlcommons.github.io/croissant/docs/croissant-rai-spec.html

File: datasets/1.0/dices-350/metadata.json (hunk 2 above)
Comment (JoanGi) [reply datasets/1.0/dices-350/metadata.json]: Checked, examples are ready

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 3]
```diff
@@ -73,26 +73,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
+ML_COMMONS_RAI_DATA_ANNOTATION_DEMOGRAPHICS = RAI.annotatorDemographics
+ML_COMMONS_RAI_DATA_ANNOTATION_TOOLS = RAI.machineAnnotationTools
 ML_COMMONS_RAI_DATA_USE_CASES = RAI.dataUseCases
 ML_COMMONS_RAI_DATA_BIASES = RAI.dataBiases
-ML_COMMONS_RAI_DATA_LIMITATION = RAI.dataLimitation
+ML_COMMONS_RAI_DATA_LIMITATION = RAI.dataLimitations
 ML_COMMONS_RAI_DATA_SOCIAL_IMPACT = RAI.dataSocialImpact
-ML_COMMONS_RAI_DATA_SENSITIVE = RAI.dataSensitive
-ML_COMMONS_RAI_DATA_MAINTENANCE = RAI.dataMaintenance
+ML_COMMONS_RAI_DATA_SENSITIVE = RAI.personalSensitiveInformation
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:92]: Two possibilities:

- Either the name of the constant should reflect the name of the property. E.g., here `ML_COMMONS_RAI_PERSONAL_SENSITIVE_INFORMATION`.
- Or you shouldn't use any constant at all and directly use `url=RAI.personalSensitiveInformation` in metadata.py.

File: python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py [hunk 4]
```diff
@@ -343,12 +329,12 @@ class Metadata(Node):
...
         url=constants.ML_COMMONS_RAI_DATA_BIASES,
     )
-    data_usecases: str | None = mlc_dataclasses.jsonld_field(
+    data_use_cases: str | None = mlc_dataclasses.jsonld_field(
         default=None,
         input_types=[SDO.Text],
         url=constants.ML_COMMONS_RAI_DATA_USE_CASES,
     )
-    data_limitation: str | None = mlc_dataclasses.jsonld_field(
+    data_limitations: str | None = mlc_dataclasses.jsonld_field(
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py:337]: Can you adapt the right cardinality: `cardinality="MANY"`

**Review by marcenacp — CHANGES_REQUESTED** [review]

File: python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py (hunk 4 above)
Comment (JoanGi) [reply python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py]: Added cardinality “many” and “one” concerning the 1.0 SPEC to any field. 

This new structure is much clearer when it comes to performing changes, congrats :)!!!.

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 5]
```diff
@@ -75,26 +75,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
-ML_COMMONS_RAI_DATA_COLLECTION_TYPE_OTHERS = RAI.dataCollectionTypeOthers
-ML_COMMONS_RAI_DATA_COLLECTION_MISSING = RAI.dataCollectionMissing
-ML_COMMONS_RAI_DATA_COLLECTION_RAW = RAI.dataCollectionRaw
-ML_COMMONS_RAI_DATA_COLLECTION_TIMEFRAME_START = RAI.dataCollectionTimeFrameStart
-ML_COMMONS_RAI_DATA_COLLECTION_TIMEFRAME_END = RAI.dataCollectionTimeFrameEnd
-ML_COMMONS_RAI_DATA_PREPROCESSING_IMPUTATION = RAI.dataPreprocessingImputation
-ML_COMMONS_RAI_DATA_PREPROCESSING_PROTOCOL = RAI.dataPeprocessingProtocol
-ML_COMMONS_RAI_DATA_PREPROCESSING_MANIPULATION = RAI.dataPreprocessingManipulation
+ML_COMMONS_RAI_DATA_COLLECTION_MISSING = RAI.dataCollectionMissingData
+ML_COMMONS_RAI_DATA_COLLECTION_RAW = RAI.dataCollectionRawData
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:79]: ML_COMMONS_RAI_DATA_COLLECTION_RAW_DATA

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 6]
```diff
@@ -75,26 +75,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
+ML_COMMONS_RAI_DATA_COLLECTION_TIMEFRAME = RAI.dataCollectionTimeFrame
+ML_COMMONS_RAI_DATA_IMPUTATION_PROTOCOL = RAI.dataPreprocessingImputation
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:81]: ML_COMMONS_RAI_DATA_PREPROCESSING_IMPUTATION

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 7]
```diff
@@ -75,26 +75,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
+ML_COMMONS_RAI_DATA_PREPROCESSING_PROTOCOL = RAI.dataPreprocessingProtocol
+ML_COMMONS_RAI_DATA_MANIPULATION_PROTOCOL = RAI.dataPreprocessingManipulation
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:83]: ML_COMMONS_RAI_DATA_PREPROCESSING_MANIPULATION

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 8]
```diff
@@ -75,26 +75,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
 ML_COMMONS_RAI_DATA_ANNOTATION_PROTOCOL = RAI.dataAnnotationProtocol
 ML_COMMONS_RAI_DATA_ANNOTATION_PLATFORM = RAI.dataAnnotationPlatform
 ML_COMMONS_RAI_DATA_ANNOTATION_ANALYSIS = RAI.dataAnnotationAnalysis
-ML_COMMONS_RAI_DATA_ANNOTATION_PER_ITEM = RAI.dataAnnotationPerItem
-ML_COMMONS_RAI_DATA_ANNOTATION_DEMOGRAPHICS = RAI.dataAnnotationDemographics
-ML_COMMONS_RAI_DATA_ANNOTATION_TOOLS = RAI.dataAnnotationTools
+ML_COMMONS_RAI_DATA_ANNOTATION_PER_ITEM = RAI.annotationsPerItem
+ML_COMMONS_RAI_DATA_ANNOTATION_DEMOGRAPHICS = RAI.annotatorDemographics
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:88]: ML_COMMONS_RAI_DATA_ANNOTATOR_DEMOGRAPHICS

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 9]
```diff
@@ -75,26 +75,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
+ML_COMMONS_RAI_DATA_ANNOTATION_TOOLS = RAI.machineAnnotationTools
 ML_COMMONS_RAI_DATA_USE_CASES = RAI.dataUseCases
 ML_COMMONS_RAI_DATA_BIASES = RAI.dataBiases
-ML_COMMONS_RAI_DATA_LIMITATION = RAI.dataLimitation
+ML_COMMONS_RAI_DATA_LIMITATION = RAI.dataLimitations
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:92]: ML_COMMONS_RAI_DATA_LIMITATIONS

File: python/mlcroissant/mlcroissant/_src/core/constants.py [hunk 10]
```diff
@@ -75,26 +75,24 @@ def ML_COMMONS(ctx) -> rdflib.Namespace:
...
 ML_COMMONS_RAI_DATA_SOCIAL_IMPACT = RAI.dataSocialImpact
-ML_COMMONS_RAI_DATA_SENSITIVE = RAI.dataSensitive
-ML_COMMONS_RAI_DATA_MAINTENANCE = RAI.dataMaintenance
+ML_COMMONS_RAI_PERSONAL_SENSITIVE_INFORMATION = RAI.personalSensitiveInformation
+ML_COMMONS_RAI_DATA_MAINTENANCE = RAI.dataReleaseMaintenancePlan
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/core/constants.py:95]: ML_COMMONS_RAI_DATA_MAINTENANCE_PLAN

**Review by marcenacp — APPROVED** [review]
A few fields have wrong names (see my comments). Otherwise LGTM :) Thanks!

**Pull request merged**